src/gf_ring.cpp

```diff
@@ -58,6 +58,108 @@ void RingModN<uint32_t>::mul_coef_to_buf(
     simd::mul_coef_to_buf(a, src, dest, len, this->_card);
 }
 
+template <>
+void RingModN<uint16_t>::butterfly_ct(
+    uint16_t coef,
+    uint16_t* buf1,
+    uint16_t* buf2,
+    size_t len) const
+{
+    unsigned ratio = ALIGN_SIZE / sizeof(*buf1);
+    size_t vec_len = len / ratio;
+    size_t last_len = len - vec_len * ratio;
+
+    // perform vector operations
+    simd::butterfly_ct(coef, buf1, buf2, vec_len, this->_card);
+
+    // for last elements, perform as non-SIMD method
+    if (last_len > 0) {
+        for (size_t i = vec_len * ratio; i < len; ++i) {
+            uint16_t a = buf1[i];
+            uint16_t b = mul(coef, buf2[i]);
+            buf1[i] = add(a, b);
+            buf2[i] = sub(a, b);
+        }
+    }
+}
+
+template <>
```

Comment:
This is a copy paste of the 16-bit version…

I wonder if we could factorize this kind of code using a type trait (`is_vectorizable`) + `std::enable_if`

Reply:
I'll handle it in #222